### PR TITLE
solution

### PR DESCRIPTION
**Pull request by uzlss**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,2 +1,51 @@
...
+            if not self.hash_table[cell_number]:
+                self.hash_table[cell_number] = [key, hash(key), value]
+                self.length += 1
+                break
+            elif (self.hash_table[cell_number][0] == key
+                    and self.hash_table[cell_number][1] == hash(key)):
+                self.hash_table[cell_number][2] = value
+                break
+            else:
+                cell_number = (cell_number + 1) % self.capacity
```
Comment (ArsenPidhoretskyi) [inline app/main.py:36]:
```suggestion
            cell_number = (cell_number + 1) % self.capacity
```

File: app/main.py [hunk 2]
```diff
@@ -1,2 +1,51 @@
...
+
+    def __getitem__(self, item: Any) -> None:
+        cell_number = hash(item) % self.capacity
+        for _ in range(self.capacity):
```
Comment (ArsenPidhoretskyi) [inline app/main.py:40]: Use also while loop

File: app/main.py [hunk 3]
```diff
@@ -1,2 +1,51 @@
...
+            if self.hash_table[cell_number] is None:
+                break
+            if (self.hash_table[cell_number][0] == item
+                    and self.hash_table[cell_number][1] == hash(item)):
+                return self.hash_table[cell_number][2]
+            else:
+                cell_number = (cell_number + 1) % self.capacity
```
Comment (ArsenPidhoretskyi) [inline app/main.py:47]:
```suggestion
           cell_number = (cell_number + 1) % self.capacity
```

File: app/main.py [hunk 4]
```diff
@@ -1,2 +1,51 @@
...
+        raise KeyError
```
Comment (ArsenPidhoretskyi) [inline app/main.py:48]:
```suggestion
        raise KeyError(item)
```

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1,2 +1,49 @@
...
+    def _resize(self) -> None:
+        self.capacity *= 2
+        self.length = 0
+        temp_table: list = self.hash_table
+        self.hash_table = [None] * self.capacity
+        for cell in temp_table:
+            if cell:
+                self[cell[0]] = cell[2]
+
+    def __setitem__(self, key: Any, value: Any) -> None:
```
Comment (vasylhnatiuk) [inline app/main.py:22]:
```suggestion
    def __setitem__(self, key: Hashble, value: Any) -> None:
```

Comment (vasylhnatiuk) [reply app/main.py:22]: key must be hashable.

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 6]
```diff
@@ -1,2 +1,49 @@
...
+                self.hash_table[cell_number] = [key, hash(key), value]
+                self.length += 1
+                break
+            elif (self.hash_table[cell_number][0] == key
+                    and self.hash_table[cell_number][1] == hash(key)):
+                self.hash_table[cell_number][2] = value
+                break
+            cell_number = (cell_number + 1) % self.capacity
+
+    def __getitem__(self, item: Any) -> None:
```
Comment (vasylhnatiuk) [inline app/main.py:37]:
```suggestion
    def __getitem__(self, item: Hashable) -> Any | None:
```

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**